components/Modal/CreateRecipe.js

```diff
@@ -65,6 +71,25 @@ class CreateRecipe extends React.Component {
   dismissErrors() {
     this.setState({ inlineError: false });
     this.setState({ showErrorName: false });
+    this.setState({ showErrorDuplicate: false });
+  }
+
+  handleErrors() {
+    this.handleErrorDuplicate();
+    this.handleErrorName();
+  }
+
+  handleErrorDuplicate() {
+    utils.apiFetch(constants.get_recipes_list)
+      .then(listdata => {
+        for (const listedName of listdata.recipes) {
+          // regex to replace a space with a '-'
+          const ourName = this.state.recipe.name.replace(/\s+/g, '-');
+          if (listedName === ourName) {
+            this.setState({ showErrorDuplicate: true });
+          }
```

Comment:
Add break here, otherwise, all of recipes will be looped.
Can we use Array.prototype.includes() to replace for..of, like this? IMO, this one is more efficient than for...of.
```javascript
listdata.recipes.includes(ourName)
```